src/uct/ib/ud/base/ud_ep.c

```diff
@@ -319,16 +319,18 @@ static void uct_ud_ep_timer(ucs_wtimer_t *self)
 
     last_send = ucs_max(ep->tx.send_time, ep->tx.resend_time);
     diff      = now - last_send;
-    if (diff > 3 * iface->tx.tick) {
-        ucs_trace("scheduling resend now: %lu last_send: %lu diff: %lu tick: %lu",
-                  now, last_send, diff, ep->tx.tick);
-        uct_ud_ep_ca_drop(ep);
-        uct_ud_ep_resend_start(iface, ep);
-    }
+    if (diff > iface->tx.tick) {
+        if (diff > 3 * iface->tx.tick) {
+            ucs_trace("scheduling resend now: %lu last_send: %lu diff: %lu tick: %lu",
+                      now, last_send, diff, ep->tx.tick);
+            uct_ud_ep_ca_drop(ep);
+            uct_ud_ep_resend_start(iface, ep);
+        }
 
-    if ((diff > iface->tx.tick) && uct_ud_ep_is_connected(ep)) {
-        /* Try to request ACK twice before going into full resend mode */
-        uct_ud_ep_ctl_op_add(iface, ep, UCT_UD_EP_OP_ACK_REQ);
+        if (uct_ud_ep_is_connected(ep)) {
+            /* Try to request ACK twice before going into full resend mode */
```

Comment:
can you pls add NAK:
`/* Try to request ACK or NAK twice before going into full resend mode */`